### PR TITLE
Bump to k8s 1.10.4 and fix FVs to work with that

### LINKED ISSUE #268: Two FVs fail with K8s v1.10.4
```
• Failure [11.977 seconds]
kube-controllers FV tests
/go/src/github.com/projectcalico/kube-controllers/tests/fv/fv_test.go:45
  should not create a workload endpoint when one does not already exist [It]
  /go/src/github.com/projectcalico/kube-controllers/tests/fv/fv_test.go:785

  Expected error:
      <*errors.StatusError | 0xc420449e60>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
              Status: "Failure",
              Message: "pods \"testpod\" is forbidden: error looking up service account default/default: serviceaccount \"default\" not found",
              Reason: "Forbidden",
              Details: {Name: "testpod", Group: "", Kind: "pods", UID: "", Causes: nil, RetryAfterSeconds: 0},
              Code: 403,
          },
      }
      pods "testpod" is forbidden: error looking up service account default/default: serviceaccount "default" not found
  not to have occurred

  /go/src/github.com/projectcalico/kube-controllers/tests/fv/fv_test.go:810
------------------------------


Summarizing 2 Failures:

[Fail] kube-controllers FV tests Pod FV tests [It] should not overwrite a workload endpoint's container ID 
/go/src/github.com/projectcalico/kube-controllers/tests/fv/fv_test.go:679

[Fail] kube-controllers FV tests [It] should not create a workload endpoint when one does not already exist 
```

### PR DESCRIPTION
**Pull request by neiljerram**
Fixes #268

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,7 +43,7 @@ OS?=$(shell uname -s | tr A-Z a-z)
 ###############################################################################
 GO_BUILD_VER ?= v0.15
 
-K8S_VERSION=v1.8.1
+K8S_VERSION?=v1.10.5
```
Comment (tomdee) [inline Makefile]: Can we stick with v1.10.4 - it's what all the other repos are using

File: tests/testutils/utils.go [hunk 2]
```diff
@@ -53,16 +54,34 @@ current-context: test-context`
...
+		fmt.Sprintf("%s", os.Getenv("HYPERKUBE_IMAGE")),
+		"/hyperkube", "controller-manager",
+		fmt.Sprintf("--master=%v:8080", apiserverIp),
+		"--min-resync-period=3m",
+		"--allocate-node-cidrs=true",
+		"--cluster-cidr=192.168.0.0/16",
+		"--v=5",
+		"--service-account-private-key-file=/private.key",
+	)
+	time.Sleep(10 * time.Second)
```
Comment (tomdee) [inline tests/testutils/utils.go]: Can we avoid a hard coded 10s sleep here?

Comment (neiljerram) [reply tests/testutils/utils.go]: Yes, of course.  This was - obviously, I thought - just a stopgap while I was asking for more high level thoughts about this change.

File: tests/fv/fv_test.go [hunk 3]
```diff
@@ -713,8 +718,11 @@ var _ = Describe("kube-controllers FV tests", func() {
 			By("updating the pod's labels to trigger a cache update", func() {
 				// Definitively trigger a pod controller cache update by updating the pod's labels
 				// in the Kubernetes API. This ensures the controller has the cached WEP with container-id-1.
+				podNow, err := k8sClient.CoreV1().Pods("default").Get(podName, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (neiljerram) [reply tests/fv/fv_test.go]: I've broken down this change into individual commits that show the sequence of error messages that I saw.  For this point, please see the error messages in https://github.com/projectcalico/kube-controllers/pull/272/commits/9f0e1baceb324812f3dad93b38c8a7eb9b229eec and https://github.com/projectcalico/kube-controllers/pull/272/commits/1b41522d9112e5dc17a9d1f8b4164eb4a9573913

[the remark this answers is not in the document]
Comment (neiljerram) [reply tests/fv/fv_test.go]: (Happy to squash this all down again if that's your preference, once we're happy that the changes here are good.)

**Comment by caseydavenport** [comment]
@neiljerram LGTM - I'd prefer if it were squashed down into a single commit though.

**Comment by neiljerram** [comment]
Sure, will do.

**Pull request merged**